### PR TITLE
Feature/test and evaluate column2vec

### PR DESCRIPTION
**Pull request by OlivieFranklova**

### CODE REVIEW

File: column2Vec/reaserch/column2Vec_re.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+def time_test(column1: pd.Series, function, key: str) -> (bool, float):
+    """
+    Computes how much time it takes to compute embeddings for column using specific function.
+    Cache are not used for this test
+    :param column1: column to test
+    :param function: function to use
+    :param key: key for cache
+    """
+    # cache.set_file("no_file.txt")
+    # cache.clear_cache()
```
Comment (Zejnilovic) [inline column2Vec/reaserch/column2Vec_re.py:87]: Commented out code either has to have a good explanation next to it or should not be there.

File: column2Vec/reaserch/column2Vec_re.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    emb1 = function(column1, model, key)
+    emb2 = function(first_half, model, key + str(1))
+    emb3 = function(second_half, model, key + str(2))
+    first_second = cosine_sim(emb3, emb2)
+    whole_second = cosine_sim(emb1, emb2)
+    whole_first = cosine_sim(emb3, emb1)
+    if first_second != 1 or whole_second != 1 or whole_first != 1:
+        # print((f"For {key}, {function}:\n whole column and first half: {whole_first}\n "
+        #        f"whole column and second half: {whole_second}\n first half"
+        #        f" and second half: {first_second}\n"))
```
Comment (Zejnilovic) [inline column2Vec/reaserch/column2Vec_re.py:117]: Commented out code either has to have a good explanation next to it or should not be there.

File: column2Vec/reaserch/generate_report.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    #                     written_column.append(column[0])
+    #                 if test_results[function][column] > 0.5:
+    #                     f_sim.write(f"{column[1]} ({test_results[function][column]}),  ")
+    #                 else:
+    #
+    #                     f_not.write(f"{column[1]} ({test_results[function][column]}),  ")
+    #             f_sim.write("\n")
+    #             f_not.write("\n")
+    #         f_sim.write("\n")
+    #         f_not.write("\n")
```
Comment (Zejnilovic) [inline column2Vec/reaserch/generate_report.py:317]: Commented out code either has to have a good explanation next to it or should not be there.

File: column2Vec/reaserch/streamlit/pages/1_map.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from streamlit_app import load_data, DATE_COLUMN
+data_load_state = st.text('Loading data...')
+# Load 10,000 rows of data into the dataframe.
+data = load_data(10000)
+# Notify the reader that the data was successfully loaded.
+data_load_state.text('Loading data...done using cache!')
+hour_to_filter = st.slider('hour', 0, 23, 17)
+filtered_data = data[data[DATE_COLUMN].dt.hour == hour_to_filter]
+st.subheader(f'Map of all pickups at {hour_to_filter}:00')
+st.map(filtered_data)
```
Comment (Zejnilovic) [inline column2Vec/reaserch/streamlit/pages/1_map.py:14]:
```suggestion
st.map(filtered_data)

```

File: config.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import os
+import certifi
+
+
+def configure():
+    print("Configuring.")
+    os.environ["REQUESTS_CA_BUNDLE"] = certifi.where()
+    os.environ['CURL_CA_BUNDLE'] = certifi.where()
+    # os.environ['REQUESTS_CA_BUNDLE'] = r"/etc/ssl/cert.pem"
+    # os.environ['REQUESTS_CA_BUNDLE'] = r"/etc/ssl/openssl.cnf"
```
Comment (Zejnilovic) [inline config.py:13]: Commented out code either has to have a good explanation next to it or should not be there.

File: .gitignore [hunk 6]
```diff
@@ -1,5 +1,167 @@
...
+# PyCharm
+#  JetBrains specific template is maintained in a separate JetBrains.gitignore that can
+#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+#  and can be added to the global gitignore or merged into this file.  For a more nuclear
+#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
+.idea
+
+# Custom for this project
+fingerprints/
+**/.DS_Store
```
[the remark this answers is not in the document]
Comment (OlivieFranklova) [reply .gitignore:167]: Fixed

File: column2Vec/README.md [hunk 7]
```diff
@@ -38,7 +43,7 @@ cache.clear_persistent_cache() # clears the cache file
 
 > Inspired by [Michael J. Mior, Alexander G. Ororbia](https://arxiv.org/pdf/1903.08621)
```
[the remark this answers is not in the document]
Comment (OlivieFranklova) [reply column2Vec/README.md:44]: Fixed

File: column2Vec/impl/Column2Vec.py [hunk 8]
```diff
@@ -22,7 +28,7 @@ class Cache:
     # enable/disable cache
     __off = False
     # default cache file
-    __file = "generated/cache.txt"
+    __file = "../generated/cache.txt"
```
[the remark this answers is not in the document]
Comment (OlivieFranklova) [reply column2Vec/impl/Column2Vec.py:31]: Fixed

File: column2Vec/reaserch/column2Vec_re.py [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+# import streamlit as st
+
+import pandas as pd
+from sentence_transformers import SentenceTransformer
+
+from column2Vec.impl.functions import get_nonnumerical_data
+from column2Vec.impl.Column2Vec import (column2vec_as_sentence, column2vec_as_sentence_clean,
+                                        column2vec_as_sentence_clean_uniq, column2vec_weighted_sum,
+                                        column2vec_sum,
+                                        column2vec_weighted_avg, column2vec_avg, cache)
```
[the remark this answers is not in the document]
Comment (OlivieFranklova) [reply column2Vec/reaserch/column2Vec_re.py:16]: Fixed

File: column2Vec/reaserch/column2Vec_re.py [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+from column2Vec.reaserch.generate_report import generate_time_report, generate_sim_report, generate_stability_report, \
+    generate_partial_column_report
```
[the remark this answers is not in the document]
Comment (OlivieFranklova) [reply column2Vec/reaserch/column2Vec_re.py:19]: Fixed

File: column2Vec/reaserch/column2Vec_re.py (hunk 1 above)
Comment (OlivieFranklova) [reply column2Vec/reaserch/column2Vec_re.py:87]: Fixed

File: column2Vec/reaserch/column2Vec_re.py (hunk 2 above)
Comment (OlivieFranklova) [reply column2Vec/reaserch/column2Vec_re.py:117]: Fixed

File: column2Vec/reaserch/generate_report.py (hunk 3 above)
Comment (OlivieFranklova) [reply column2Vec/reaserch/generate_report.py:317]: Fixed

File: column2Vec/reaserch/streamlit/pages/1_map.py (hunk 4 above)
Comment (OlivieFranklova) [reply column2Vec/reaserch/streamlit/pages/1_map.py:14]: Fixed

File: config.py (hunk 5 above)
Comment (OlivieFranklova) [reply config.py:13]: Fixed

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**